packages/dx-react-grid/docs/reference/column-chooser.md

```diff
@@ -1,21 +1,51 @@
-# ColumnChooser Reference
+# ColumnChooser Plugin Reference
 
-ColumnChooser is a component that allows a user to show and hide grid columns at runtime. The component contains a column list with checkboxes that control each column's visibility. Use the Grid component's [TableColumnVisiblity](table-column-visibility.md) plugin that manages column visibility if you need to associate the ColumnChooser with a Grid.
+ColumnChooser is a plugin that allows a user to show and hide grid columns at runtime. The plugin contains a column list with checkboxes that control each column's visibility.
 
 ## User reference
 
+### Dependencies
+
+- [TableColumnVisiblity](table-column-visibility.md)
+- [Toolbar](toolbar.md)
+
 ### Properties
 
 Name | Type | Default | Description
 -----|------|---------|------------
-columns | Array&lt;[Column](grid.md#column)&gt; | | Columns that ColumnsChooser operates on.
-hiddenColumns | Array&lt;string&gt; | [] | An array containing hidden column names.
-onHiddenColumnsChange | (nextHiddenColumns: Array&lt;string&gt;) => void | | Handles column visibility changes.
+overlayComponent | ElementType&lt;[ColumnChooserOverlayProps](#columnchooseroverlayprops)&gt; | | A component that renders the column chooser overlay.
+toggleButtonComponent | ElementType&lt;[ColumnChooserToggleButtonProps](#columnchoosertogglebuttonprops)&gt; | | A component that renders the column chooser toggle button.
 containerComponent | ElementType&lt;[ColumnChooserContainerProps](#columnchoosercontainerprops)&gt; | | A component that renders the column chooser container.
 itemComponent | ElementType&lt;[ColumnChooserItemProps](#columnchooseritemprops)&gt; | | A component that renders a column chooser item.
+messages | object | | An object that specifies the [localization messages](#localization-messages).
 
 ## Interfaces
 
+### ColumnChooserOverlayProps
+
+Describes properties passed to a component that renders the column chooser overlay.
+
+A value with the following shape:
+
+Field | Type | Description
+------|------|------------
+visible | boolean | Specifies whether an overlay is visible.
+target | React element | The React element that is used to set the position of the overlay.
+onHide | () => void | An event that initiates overlay hiding.
+children | Array&lt;ReactElement&gt; | React elements used to render overlay content.
+
+### ColumnChooserToggleButtonProps
+
+Describes properties passed to a component that renders the column chooser toggle button.
+
+A value with the following shape:
+
+Field | Type | Description
+------|------|------------
+onToggle | () => void | An event that initiates overlay showing or hiding.
+getMessage | ([messageKey](#localization-messages): string) => string | Returns title for the column chooser toggle button. Available in the "@devexpress/dx-react-grid-material-ui" package.
+buttonRef | (node) => void | A function that take a React element of the root element.
```

Comment:
@kvet what will you say about this description?
Maybe (React element) => void ?

Reply:
Let's define it like this: `(ref: ReactElement) => void`